app/models/pacbio/pool.rb

```diff
@@ -16,24 +16,52 @@ class Pool < ApplicationRecord
     # after that is defined
     include DualSourcedPool
 
-    validates :libraries, presence: true
     validates_with TagValidator
 
     validates :volume, :concentration,
               :insert_size, :template_prep_kit_box_barcode, presence: true, on: :run_creation
     validates :volume, :concentration,
               :insert_size, numericality: { greater_than_or_equal_to: 0, allow_nil: true }
+    validates :libraries, presence: true
+    validates :used_aliquots, presence: true
+
+    # List of common attributes between libraries and used aliquots
+    COMMON_ATTRIBUTES = %w[volume concentration insert_size
+                           template_prep_kit_box_barcode tag_id].freeze
 
     def library_attributes=(library_options)
```

Comment:
library_attributes and used_aliquot_attributes are very similar. Maybe we can do a bit of abstraction. Also does this need to be wrapped in a transaction?

Reply:
Yeah they are similar but use different methods and directly affect different properties which makes it a bit awkward to refactor. They don't need to be in a transaction as when used in the resource they are automatically wrapped in one.